controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -78,18 +79,23 @@
 import io.pravega.shared.protocol.netty.WireCommands;
 import java.io.Serializable;
 import java.time.Duration;
+import java.util.AbstractMap;
 import java.util.ArrayList;
 import java.util.Comparator;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
+import java.util.TreeMap;
 import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionException;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
```

Comment:
Using updateStream() API when a user tries to switch the Retention Policy for an existing Stream( Stream created prior to 0.9) from non-CBR to CBR, we need to invoke `createSubscribersRecordIfAbsent` to make sure the SubscribersSet Record is created in the Stream metadata table. Without this add/createSubscriber would fail for existing Streams.

Reply:
fixed